File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int ret = 0;
        int temp = num;
        if(temp == 1) ret = 0;
        else {
            while(temp != 1){
                if(temp % 2 == 0) temp /= 2;
                else temp = (temp * 3) + 1;                
                ret++;
                if(ret > 480){
                    ret = -1;
                    break;
                }
            }    
        }
        
        return ret;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    cnt = 0
    while cnt < 500 and num != 1:
        if num % 2 == 0:
            num //= 2
        else:
            num = num * 3 + 1
        cnt += 1
    return cnt if num == 1 else -1